build/README.md

```diff
@@ -1,3 +1,104 @@
-# Build and Distribution
+# Building Rook
 
-This directory contains build, test and distribution scripts.
\ No newline at end of file
+While it is possible to build Rook directly on your host, we highly discourage it.
+Rook has many dependencies (mostly through Ceph) and it might be hard to keep
+up with the different versions across distros. Instead we recommend that you use
+the build process that runs inside a Docker container. It results in a consistent
+build, test and release environment.
+
+## Requirements
+
+A capable machine with Docker installed locally:
+
+  * MacOS: you can use Docker for Mac or you're own docker-machine
+  * Linux: any distro with a recent version of Docker would work
+
+We do not currently support building on a remote docker host.
+
+## Using the Build Container
+
+To run a command inside the container:
+
+```
+> build/run <command>
+```
+
+This will run  `<command>` from inside the container where all tools and dependencies
+have been installed and configured. The current directory is set to the source directory
+which is bind mounted (or rsync'd on some platforms) inside the container.
+
+The first run of `build/run` will pull the container (which can take a while).
+
+If you don't pass a command you get an interactive shell inside the container.
+
+```
+> build/run
+rook@moby:~/go/src/github.com/rook/rook$
+```
+
+## Building
+
+Now you can build in the container normally, like so:
+
+```
+build/run make -j4
```

Comment:
if this is run in the container directly would it be `make -j4`? or maybe a comment would clarify that the rest of the examples are actually called outside the containter